soh/soh/Enhancements/randomizer/3drando/shops.cpp

```diff
@@ -164,6 +160,12 @@ int GetRandomShopPrice() {
         max = 199; // 995/5
     }
     if (max != 0) {
+        if (Settings::ShopsanityPricesAffordable.Is(true)) {
+            if(max == 19) {
+                return 10;
+            }
```

Comment:
do we need this? shouldn't this already happen based on the logic in `GetPriceAffordable`?

Reply:
I did it this way so that there was no need to go into that function and do the array copying that goes on in there. It's just less resource intensive.

Reply:
If you think the consistency in execution would be better, it should work the other way, too.

Reply:
why not just throw
```cpp
if (Settings::ShopsanityPrices.Is(RO_SHOPSANITY_PRICE_STARTER)) {
    return 10;
}
```
at the top of `GetPriceAffordable`?

checking the max here just makes it hard to know why this is happening.

Reply:
the last block could even be
```cpp
if (Settings::ShopsanityPrices.IsNot(RO_SHOPSANITY_PRICE_BALANCED) {
    return Settings::ShopsanityPricesAffordable.Is(true) ? GetPriceAffordable() : GetPriceFromMax(max);
}
```

Reply:
Well, we can't return 10 like that because we don't know if Affordable is checked there or not. Max of 19 still plays a part if starter is selected and Affordable isn't. The better option would be what you posted for `if (max != 0)`, if you want unified functionality. But either we check for `max == 19` where it is, or we check for `Settings::ShopsanityPricesAffordable.Is(true)` both in the starter check AND max != 0 if we want to keep the codeflow out of `GetPriceAffordable()` when it doesn't need to randomly select from a single element.

Reply:
> Well, we can't return 10 like that because we don't know if Affordable is checked there or n

in `GetPriceAffordable`? Yes we do know Affordable is checked.

Reply:
i'm saying

```cpp
int GetPriceAffordable() {
   if (Settings::ShopsanityPrices.Is(RO_SHOPSANITY_PRICE_STARTER)) {
     return 10;
   }

    static const std::vector<int> affordablePrices = { 10, 105, 205, 505 };
    std::vector<int> priceList;
    uint8_t maxElements = Settings::ShopsanityPrices.Value<uint8_t>();
    for (int i = 0; i < maxElements; i++) {
        priceList.push_back(affordablePrices.at(i));
    }
    return RandomElement(priceList);
}
```

and then where you currently have `if (max != 0)`
```cpp
if (Settings::ShopsanityPrices.IsNot(RO_SHOPSANITY_PRICE_BALANCED) {
    return Settings::ShopsanityPricesAffordable.Is(true) ? GetPriceAffordable() : GetPriceFromMax(max);
}
```

Reply:
Oh, I see. I misread where you wanted to put that check. Yeah, that would probably work.